addons/product/controllers/catalog.py

```diff
@@ -40,6 +40,8 @@ def product_catalog_update_order_line_info(self, res_model, order_id, product_id
                  the quantity selected.
         :rtype: float
         """
+        if (context := kwargs.get('context')):
+            request.update_context(**context)
```

Comment:
Why do you need context here? 
Can't you override the catalog on your side to get all the args you need and explicitly pass them to `create`?

I don't like this approach, it will be a nightmare to debug because the creation will magically use the context values, and if someone, someday, create a record in sudo, it might allow to give parameters we don't want.